utils/data_cleaning.py

```
@@ -41,30 +372,558 @@ def robust_sort_key(x: Any) -> tuple:
+        if validate_quality:
+            quality_report = validate_data_quality(df_cleaned)
+            cleaning_report['quality_report'] = quality_report
+            cleaning_report['cleaning_steps'].append("Data quality validation")
+        
+        cleaning_report['cleaning_steps'].append("Cleaning complete")
+        
+        logger.info(f"Data cleaning complete: {df.shape} -> {df_cleaned.shape}")
+        return df_cleaned, cleaning_report
+        
+    except Exception as e:
+        logger.error(f"Data cleaning failed: {e}")
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)